noir-projects/aztec-nr/aztec/src/state_vars/shared_mutable/scheduled_value_change.nr

```diff
@@ -0,0 +1,279 @@
+            //
+            //      block of      historical        
+            //       change         block            block horizon
+            //   =======|=============N===================H===========>
+            //                         ^                   ^
+            //                         ---------------------
+            //                                delay
+
+            historical_block_number + DELAY
+        } else {
+            // If the block of change has not yet been mined however, then there's two possible scenarios.
+            //   a) It could be so far into the future that the block horizon is actually determined by the delay,
+            //      because a new change could be scheduled and take place _before_ the currently scheduled one. This is
+            //      similar to the scenario where the block of change is in the past: the time horizon is the block
+            //      prior to the earliest one in which a new block of change might land.
+            // 
+            //         historical                         
+            //           block                        block horizon      block of change
+            //        =====N=================================H=================|=========>
+            //              ^                                 ^
+            //              |                                 |   
+            //              -----------------------------------
+            //                           delay
+            //
+            //   b) It could be fewer than `delay` blocks away from the historical block number, in which case it would
+            //      become the limiting factor for the time horizon, which would be the block right before the block of
+            //      change (since by definition the value changes at the block of change).
+            //
+            //           historical                         block horizon
+            //             block     block of change       if not scheduled
+            //        =======N=============|===================H=================>
+            //                ^           ^                    ^
+            //                |     actual horizon             |   
+            //                ----------------------------------
+            //                             delay
+            //      
+            // Note that the current implementation does not allow the caller to set the block of change to an arbitrary
+            // value, and therefore scenario a) is not currently possible. However implementing #5501 would allow for 
+            // this to happen.
+
+            // Because historical_block_number < self.block_of_change, then block_of_change > 0 and we can safely
+            // subtract 1.
+            min(self.block_of_change - 1, historical_block_number + DELAY)
```

Comment:
Just thinking about the -1. So we are having the -1 to make sure that the block horizon ends before the change occurs. 

But probably be nicce to have some better clarity on why it is not needed for the `historical_block_number + DELAY`. 

Like, If I m just thinking of an execution where you havae 
```
get_block_horizon(current_block_number) <-- horizon would be current + delay
schedule_change(new, current_block_number) <-- block_of_change would be current + delay
```

Which sounds weird?

But as we are constraining the functions in the `shared_mutable` to only use the `get_block_horizon` in private, you cannot really get this case, as the current cannot be used. 

---
Rambly rambler